soccer/src/soccer/strategy/agent/position/goalie.cpp

```diff
@@ -35,8 +35,19 @@ Goalie::State Goalie::update_state() {
         return CLEARING;
     }
 
-    // otherwise, default to idling
-    return IDLING;
+    rj_geometry::Point robot_position = world_state->get_robot(true, robot_id_).pose.position();
+    double distance_to_ball = robot_position.dist_to(ball_pt);
+    if (latest_state_ == PASSING) {
+        if (!ball_in_box) {
+            return IDLING;
+        }
+    } else if (latest_state_ == RECEIVING) {
+        if (distance_to_ball < 0.1) {
```

Comment:
this should use the `check_is_done()` method from Position superclass, which links the notion of done-ness directly to the planner (and prevents strategy from having to know how planners are implemented). I think that's why you're seeing poor behavior from the receiving robot.

see here: https://github.com/RoboJackets/robocup-software/blob/ros2/soccer/src/soccer/strategy/agent/position/offense.cpp#L9

Reply:
not sure if this is how you implemented it in the other Positions, but double-check